### DISCUSSION TITLE
Action not being triggered (push, not to master)

### DISCUSSION THREAD

**Discussion by spaceemotion** [community/community#26486] [Actions]
<p>I just wanted to add the first github action to our repo, and since we work with PRs, I can’t push to master directly. I’ve created the following config under “.github/workflows/validate.yml”:</p>
<pre><code>name: Valdiate OpenAPI

on:
  push:
    paths:
    - "openapi/**"

jobs:
  validate:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-node@v1
        with:
          node-version: 12.x

      - name: Obtain NPM Cache directory
        id: node-cache-dir
        run: echo "::set-output name=dir::$(yarn cache dir)"

      - name: Cache NPM dependencies
        uses: actions/cache@v1
        with:
          path: ${{ steps.node-cache-dir.outputs.dir }}
          key: ${{ runner.os }}-node-${{ hashFiles('**/yarn.lock') }}
          restore-keys: ${{ runner.os }}-node-

      - run: npx swagger-cli validate openapi/openapi.yml
</code></pre>
<p>The only problem I have is that GitHub is not picking up my config at all, there are no checks running whatsoever (even before I added the “path” entry under the push trigger, so it should have ran most definitely).</p>
<p>Is this something that’ll only be picked up once it gets merged into master? Thanks in advance!</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/spaceemotion">@spaceemotion</a> , </p>
<p>Do you use a public repo or a private repo? If it is a public one, would you mind sharing the repo link here ? </p>
<p>For a workflow in non-master branch (for example feature branch) to be triggered, please make sure the workflow yml file exists in feature branch. </p>

**Comment by spaceemotion** [comment]
<p>Hey <a class="mention" href="/yanjingzhu">@yanjingzhu</a> thanks for the reply,</p>
<p>this is in a private repo in our organization. I’ve made a new branch from master (let’s call it “validate-api”), and added the workflow file to that using a new commit. Pushed that to github and made a PR for master.</p>
<p>However, since that was during the outage just yesterday, I thought it might have failed due to that. I’ve pushed some minor changes (typo fixes) after the issues got resolved, but it still did not get picked up.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Can you see validate-api branch in github web portal now?</p>
<p>Please go to the code page of validate-api branch , would you mind removing the paths filter for on: push event, and then make a change and push to validate-api branch? Is there a new workflow run be triggered on validate-api branch? </p>

**Comment by spaceemotion** [comment]
<p>Weird, I actually did try the same thing just yesterday, but it might just have been during the outage. I removed the filter and it seems to work now. Will add it back before merging then. Cheers!</p>
